runtime/taskcontext.go

```diff
@@ -80,25 +84,38 @@ type TaskContextController struct {
 }
 
 // NewTaskContext creates a TaskContext and associated TaskContextController
-func NewTaskContext(tempLogFile string, task TaskInfo) (*TaskContext, *TaskContextController, error) {
+func NewTaskContext(tempLogFile string,
+	task TaskInfo,
+	server webhookserver.WebHookServer) (*TaskContext, *TaskContextController, error) {
 	logStream, err := stream.New(tempLogFile)
 	if err != nil {
 		return nil, nil, err
 	}
 	ctx := &TaskContext{
-		logStream: logStream,
-		TaskInfo:  task,
+		logStream:   logStream,
+		logLocation: tempLogFile,
+		TaskInfo:    task,
+		webHookSet:  webhookserver.NewWebHookSet(server),
 	}
 	return ctx, &TaskContextController{ctx}, nil
 }
 
 // CloseLog will close the log so no more messages can be written.
 func (c *TaskContextController) CloseLog() error {
+	if c.logClosed {
+		return nil
+	}
+
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	c.logClosed = true
 	return c.logStream.Close()
```

Comment:
hmm... I have mixed feeling here..
Ie. should we panic if we try to close twice, or always allow multiple close calls..  I guess this is easier :)